pageserver/src/walredo.rs

```diff
@@ -414,7 +414,8 @@ impl PostgresRedoManager {
                     "ClearVisibilityMapFlags record on unexpected rel {}",
                     rel
                 );
-                if let Some(heap_blkno) = *new_heap_blkno {
+                for heap_blkno in [heap_blkno_1, heap_blkno_2].into_iter().flatten() {
+                    let heap_blkno = *heap_blkno;
```

Comment:
I'm OK with this, but I think a comment would be in order. If I saw this without the old code, I'd have to look up the rust docs of `flatten()` and think for a while to figure out what this does. Something like;
```
// One ClearVisibilityMapFlags record can contain up to two independent changes that apply to the same VM page, stored in `new_heap_blkno` and `old_heap_bkno` fields. The fields that are not used are None. Apply any changes from the fields that are not None.
```
(ok, that's a bit verbose and not very clear either.. Maybe you can come up with something better)

Or maybe it's better to just repeat the code snippet. That might be most readable after all.